internal/resources/provisioner/resource_provisioner.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package provisioner
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	provisioner "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/provisioner"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+type (
+	contextMethodKey struct{}
+)
+
+func ResourceProvisioner() *schema.Resource {
+	return &schema.Resource{
+		Schema:        provisionerSchema,
+		ReadContext:   dataSourceProvisionerRead,
+		CreateContext: resourceProvisionerCreate,
+		UpdateContext: resourceProvisionerInPlaceUpdate,
+		DeleteContext: resourceProvisionerDelete,
+		Description:   "Tanzu Mission Control Provisioner Resource",
+	}
+}
+
+var provisionerSchema = map[string]*schema.Schema{
+	NameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the provisioner",
+		Required:    true,
+		ForceNew:    true,
+	},
+	ManagementClusterNameKey: {
+		Type:        schema.TypeString,
+		Description: "Name of the management cluster",
+		Required:    true,
+		ForceNew:    true,
+	},
+	OrgIDKey: {
+		Type:        schema.TypeString,
+		Description: "ID of the organization",
+		Optional:    true,
+	},
+	common.MetaKey: common.Meta,
+}
+
+func resourceProvisionerCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config, ok := m.(authctx.TanzuContext)
+
+	if !ok {
+		log.Println("[ERROR] error while retrieving Tanzu auth config")
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	provisionerName, ok := d.Get(NameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read provisioner name")
+	}
+
+	managementClusterName, ok := d.Get(ManagementClusterNameKey).(string)
```

Comment:
Replace it with converter

Reply:
The schema for the provisioner is straight forward and is not nested as other resources. Adding the converter is not required which is comparatively more lines of changes than the current implementation, which is very easy to read and understand

Reply:
I don't think its true, converter would not add more lines of code, and while we have the existing converter logic that can be utilised make it a practice to use it instead of relying on the legacy method. This will also ensure that we use and enhance the converter for future use cases

Reply:
We have to read provisioner and management cluster name alone from schema to pass it to the request. This is done as part of the changes in the code, adding the converter involves adding the converter map for the model and construct functions for the requests and the response.

Converter is useful in case of nested schemas where we have to construct and flatten the schema. Using converter in such case, helps avoid multi lines of code written. Which is not the case with provisioner resource

Reply:
Making use of converter